mobile-academy/src/main/java/org/motechproject/nms/mobileacademy/service/impl/MobileAcademyServiceImpl.java

```diff
@@ -457,4 +470,60 @@ public String getScoresForUser(Long callingNumber) {
 
         return scores;
     }
-}
+
+    @Override
+    public void updateMsisdn(Long oldCallingNumber, Long newCallingNumber) {
+
+        if (oldCallingNumber.equals(newCallingNumber)) {
+            return;
+        }
+        // Update Msisdn  In MTRAINING_MODULE_BOOKMARK
+        LOGGER.debug("Fetching Bookmarks for Msisdn {}.", oldCallingNumber);
+        List<Bookmark> existingBookmarks = bookmarkService.getAllBookmarksForUser(oldCallingNumber.toString());
+        if (existingBookmarks.size() > 0) {
+            int i;
+            Bookmark bookmark;
+            for (i = 0; i < existingBookmarks.size(); i++) {
+                bookmark = existingBookmarks.get(i);
+                bookmark.setExternalId(newCallingNumber.toString());
+                bookmarkService.updateBookmark(bookmark);
+            }
+            LOGGER.debug("Updated MSISDN {} to {} in {} Bookmarks", oldCallingNumber, newCallingNumber, i);
+        } else {
```

Comment:
For here and below, how about checks first to see if the new phone number has existing bookmarks and completion records already? What to do then?